src/VisualStudio/CSharp/Impl/Debugging/LocationInfoGetter.cs

```diff
@@ -1,94 +1,155 @@
+                    var token = type.GetNameToken();
+                    return token.IsMissing ? missingInformationPlaceholder : token.Text;
+                },
+                type => type.GetAncestor<TypeDeclarationSyntax>());
 
-                    var accessor = token.GetAncestor<AccessorDeclarationSyntax>();
-                    var memberLine = accessor == null
-                        ? text.Lines.GetLineFromPosition(memberDecl.SpanStart).LineNumber
-                        : text.Lines.GetLineFromPosition(accessor.SpanStart).LineNumber;
+            // containing namespace(s)
+            var containingNamespace = (containingType ?? memberDeclaration).GetAncestor<NamespaceDeclarationSyntax>();
+            AppendContainingDeclarationNames(
+                ref containingDeclarationNames,
+                ref containingNamespace,
+                @namespace =>
+                {
+                    var syntax = @namespace.Name;
+                    return syntax.IsMissing ? missingInformationPlaceholder : syntax.ToString();
+                },
+                @namespace => @namespace.GetAncestor<NamespaceDeclarationSyntax>());
 
-                    name = memberSymbol.ToDisplayString(s_nameFormat);
-                    lineOffset = lineNumber - memberLine;
-                    return new DebugLocationInfo(name, lineOffset);
+            var pooled = PooledStringBuilder.GetInstance();
+            var builder = pooled.Builder;
+            if (containingDeclarationNames != null)
+            {
+                for (var i = containingDeclarationNames.Count - 1; i >= 0; i--)
+                {
+                    builder.Append(containingDeclarationNames[i]);
+                    builder.Append('.');
                 }
+                containingDeclarationNames.Free();
             }
 
-            return default(DebugLocationInfo);
+            // simple name
+            var nameToken = fieldDeclaratorOpt?.Identifier ?? memberDeclaration.GetNameToken();
+            if (nameToken.IsMissing)
+            {
+                builder.Append(missingInformationPlaceholder);
+            }
+            else if (nameToken == default(SyntaxToken))
+            {
+                Debug.Assert(memberDeclaration.Kind() == SyntaxKind.ConversionOperatorDeclaration);
+                builder.Append((memberDeclaration as ConversionOperatorDeclarationSyntax)?.Type);
+            }
+            else
+            {
+                if (memberDeclaration.Kind() == SyntaxKind.DestructorDeclaration)
+                {
+                    builder.Append('~');
+                }
+                builder.Append(nameToken.Text);
+            }
+
+            // parameter list (if any)
+            builder.Append(memberDeclaration.GetParameterList());
+
+            return pooled.ToStringAndFree();
+        }
+
+        private static void AppendContainingDeclarationNames<TDeclarationSyntax>(
```

Comment:
As we discussed offline, I think this could be more clearly expressed as a single while loop walking up the syntax tree.  You can use an ```ArrayBuilder``` as a stack to avoid allocations.